File: crates/kb-code-server/tests/fixtures/oracle/shadow.ts
// Oracle fixture: param/block/function-hoist shadowing traps (TypeScript).

const value = 1;

function run(value: number): number {
  // ref `value` → param (not module const)
  return value;
}

function blockShadow(): void {
  let x = 10;
  {
    let x = 20;
    // ref `x` → inner let
    const y = x;
  }
}

function hoistCall(): number {
  // function decl is hoisted in scope → later()
  return later();
  function later(): number {
    return 1;
  }
}

function letNotBefore(): void {
  // `later` is a let — NOT visible before declaration → unbound
  const y = later;
  let later = 1;
}

function unboundUse(): void {
  const m = missingName;
}

File: crates/kb-code-server/tests/fixtures/oracle/shadow.rs
// Oracle fixture: param/block/loop shadowing traps (Rust).
// Dense same-name bindings; golden rows pin expected def sites.

fn value() -> i32 {
    1
}

fn run(value: i32) -> i32 {
    // ref `value` → param on this fn (not module fn)
    value
}

fn block_shadow() {
    let x = 10;
    {
        let x = 20;
        // ref `x` → inner let
        let _y = x;
    }
}

fn loop_shadow(item: i32) {
    for item in 0..3 {
        // ref `item` → loop binding (not param)
        let _z = item;
    }
    // ref `item` after loop → param (for-scope does not leak under our model)
    let _w = item;
}

fn unbound_use() {
    // deliberately unbound
    let _m = missing_name;
}
